### PR TITLE
logs: hide timestamp if the program is run not in TTY

### LINKED ISSUE #3358: logs: hide timestamp by default
while prepending log lines with timestamp may be a good feature, it must be disabled by default, the same way as all other programs do

### PR DESCRIPTION
**Pull request by AliceInHunterland**
Close #3358

### CODE REVIEW

**Comment by roman-khimov** [comment]
We need this to be dependent on tty.

File: cli/options/options.go [hunk 1]
```diff
@@ -239,7 +241,11 @@ func HandleLoggingParams(debug bool, cfg config.ApplicationConfiguration) (*zap.
 	cc.DisableStacktrace = true
 	cc.EncoderConfig.EncodeDuration = zapcore.StringDurationEncoder
 	cc.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
-	cc.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	if term.IsTerminal(int(os.Stdout.Fd())) {
+		cc.EncoderConfig.EncodeTime = zapcore.TimeEncoderOfLayout("")
+	} else {
+		cc.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	}
```
Comment (AnnaShaleva) [inline cli/options/options.go:248]: It should be vice-versa. If we use TTY (usual I do this as a developer), then we need to see the time.

File: cli/options/options.go [hunk 2]
```diff
@@ -239,7 +241,11 @@ func HandleLoggingParams(debug bool, cfg config.ApplicationConfiguration) (*zap.
 	cc.DisableStacktrace = true
 	cc.EncoderConfig.EncodeDuration = zapcore.StringDurationEncoder
 	cc.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
-	cc.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+	if term.IsTerminal(int(os.Stdout.Fd())) {
+		cc.EncoderConfig.EncodeTime = zapcore.TimeEncoderOfLayout("")
```
Comment (AnnaShaleva) [inline cli/options/options.go:245]: Zap doesn't support an empty time encoder, so it's a little hack and this code will produce indented logs. @roman-khimov, @532910, are we OK with the following output?
```
	INFO	MaxBlockSize is not set or wrong, setting default value	{"MaxBlockSize": 262144}
	INFO	MaxBlockSystemFee is not set or wrong, setting default value	{"MaxBlockSystemFee": 900000000000}
	INFO	MaxTransactionsPerBlock is not set or wrong, using default value	{"MaxTransactionsPerBlock": 512}
	INFO	MaxValidUntilBlockIncrement is not set or wrong, using default value	{"MaxValidUntilBlockIncrement": 5760}
	INFO	restoring blockchain	{"version": "0.2.12"}
```

**Review by AnnaShaleva — COMMENTED** [review]
@AliceInHunterland, fix the commit message and PR title. It's not "by default".

File: cli/options/options.go (hunk 2 above)
Comment (532910) [reply cli/options/options.go:245]: no

Comment (AnnaShaleva) [reply cli/options/options.go:245]: @AliceInHunterland, can we remove the starting indent? From what I searched, we can't.

Comment (AliceInHunterland) [reply cli/options/options.go]: We can do this:

```
INFO    initial gas supply is not set or wrong, setting default value   {"InitialGASSupply": "52000000"}
INFO    MaxBlockSize is not set or wrong, setting default value {"MaxBlockSize": 262144}
INFO    MaxBlockSystemFee is not set or wrong, setting default value    {"MaxBlockSystemFee": 900000000000}
INFO    MaxTransactionsPerBlock is not set or wrong, using default value        {"MaxTransactionsPerBlock": 512}

```

**Review by 532910 — APPROVED** [review]
lgtm

**Review by AnnaShaleva — APPROVED** [review]

**Pull request merged**